source/world_builder/features/subducting_plate_models/composition/water_content.cc

```diff
@@ -0,0 +1,231 @@
+          this->world = world_;
+          this->name = "water content";
+        }
+
+        WaterContent::~WaterContent()
+          = default;
+
+        void
+        WaterContent::declare_entries(Parameters &prm, const std::string & /*unused*/)
+        {
+          // Document plugin and require entries if needed.
+          // Add compositions to the required parameters.
+          prm.declare_entry("", Types::Object({"compositions"}),
+                            "WaterContent compositional model. Sets constant compositional field.");
+
+          // Declare entries of this plugin
+          prm.declare_entry("min distance slab top", Types::Double(0),
+                            "todo The depth in meters from which the composition of this feature is present.");
+          prm.declare_entry("max distance slab top", Types::Double(std::numeric_limits<double>::max()),
+                            "todo The depth in meters to which the composition of this feature is present.");
+          prm.declare_entry("density", Types::Double(3000.0),
+                            "The reference density used for determining the lithostatic pressure for calculating "
+                            "the bound water content.");
+          prm.declare_entry("compositions", Types::Array(Types::UnsignedInt(),0),
+                            "A list with the labels of the composition which are present there.");
+          prm.declare_entry("lithology",  Types::String("peridotite"),
+                            "The lithology used to determine which polynomials to use for calculating the water content.");
+          prm.declare_entry("initial water content", Types::Double(5),
+                            "The value of the initial water content (in wt%) for the lithology at the trench. This represents the "
+                            "max value applied to this lithology.");
+          prm.declare_entry("operation", Types::String("replace", std::vector<std::string> {"replace", "replace defined only", "add", "subtract"}),
+                            "Whether the value should replace any value previously defined at this location (replace) or "
+                            "add the value to the previously define value. Replacing implies that all compositions not "
+                            "explicitly defined are set to zero. To only replace the defined compositions use the replace only defined option.");
+
+        }
+
+        void
+        WaterContent::parse_entries(Parameters &prm)
+        {
+          min_depth = prm.get<double>("min distance slab top");
+          max_depth = prm.get<double>("max distance slab top");
+          density = prm.get<double>("density");
+          compositions = prm.get_vector<unsigned int>("compositions");
+          max_water_content = prm.get<double>("initial water content");
+          operation = string_operations_to_enum(prm.get<std::string>("operation"));
+          lithology_str = prm.get<std::string>("lithology");
```

Comment:
You have an enum, there is no need to store a string, since you are only really need the enum afterward. So I would suggest to convert it directly to an enum here and just pass that around. This way you only have to do a string comparison once at startup and not every time you call the plugin.